src/msw/uilocale.cpp

```diff
@@ -209,9 +475,30 @@ class wxUILocaleImplName : public wxUILocaleImpl
     {
         // Getting the locale name seems to be the simplest way to see if it's
         // really supported: unknown locale result in an error here.
+        //
+        // The latter may have been a valid assumption for Windows 7 and below,
+        // but at least for Windows 10 the function only fails if the given
+        // locale name is not a valid BCP 47 identifier. For example,
+        // valid language codes can have 2 or 3 letters:
+        // - using name "w" fails
+        // - using name "wx" succeeds
+        // - using name "wxy" succeeds
+        // - using name "wxyz" fails
         if ( !ms_GetLocaleInfoEx(name, LOCALE_SNAME, NULL, 0) )
             return NULL;
 
+        // It can be checked whether the locale is "constructed" or not.
+        // "not constructed" means the locale is a predefined locale,
+        // "constructed" means the locale is not predefined, but has to be constructed.
+        // For example, "de-US" would be a constructed locale.
+        // Using LOCALE_ICONSTRUCTEDLOCALE to query the locale status is discouraged
+        // by Microsoft (the constant is not even defined in a Windows header file).
+        // However, for now constructed locales will be rejected here.
+        int isConstructed = 0;
+        if (!ms_GetLocaleInfoEx(name, LOCALE_ICONSTRUCTEDLOCALE | LOCALE_RETURN_NUMBER,
+                                (LPWSTR)&isConstructed, sizeof(int)) || isConstructed != 0)
```

Comment:
While testing, I've discovered that this check fails for `en-GB` locale under Windows 7 (but succeeds under Windows 10), even though the locale seems to be supported just fine by the system. So I'm really not sure if we should be using this.

Reply:
Finally, I believe we should use this under Windows 10 _only_, but not under Windows 7 (not sure about the intermediate versions).